src/ray/object_manager/common.cc

```diff
@@ -134,7 +134,7 @@ Status PlasmaObjectHeader::WriteAcquire(
   RAY_CHECK(sem.header_sem);
 
   RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.object_sem, timeout_point));
-  RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.header_sem, timeout_point));
+  RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.header_sem));
```

Comment:
Should this be applied only when `timeout == 0`?

Reply:
Hmm good question. I feel like in general header acquire time is very short (and it shouldn't take forever), so I feel like it is probably better having static timeout. For example, the same issue can happen when timeout is 0.0000001 (though it is practically not happening).